repomgr/repomgr.go

```diff
@@ -527,10 +554,112 @@ func (rm *RepoManager) HandleExternalUserEvent(ctx context.Context, pdsid uint,
 		return fmt.Errorf("opening external user repo (%d, root=%s): %w", uid, root, err)
 	}
 
-	repoDid := r.RepoDid()
+	if r.DataCid() != olddc {
+		return ErrUncleanRebase
+	}
 
-	if did != repoDid {
-		return fmt.Errorf("DID in repo did not match (%q != %q)", did, repoDid)
+	if err := rm.CheckRepoSig(ctx, r, did); err != nil {
+		return err
+	}
+
+	// TODO: this is moderately expensive and currently results in the users
+	// entire repo being held in memory
```

Comment:
we probably want/need some global cap on overall repo size anways

Reply:
yeah thats gonna be a spicy discussion